src/offset/local/windows_sys/mod.rs

```diff
@@ -0,0 +1,151 @@
+use std::{fmt, io::Error, ptr, result::Result};
+
+use winapi::shared::minwindef::*;
+use winapi::um::minwinbase::SYSTEMTIME;
+use winapi::um::timezoneapi::*;
+
+/// This macro calls a Windows API FFI and checks whether the function errored with the provided error_id. If an error returns,
+/// the macro will return an `Error::last_os_error()`.
+///
+/// # Safety
+///
+/// This macro can only check one idea, and provided error ID must be the corresponding error ID.
+macro_rules! windows_sys_call {
+    ($name:ident($($arg:expr),*), $error_id:expr) => {
+        if $name($($arg),*) == $error_id {
+            return Err(Error::last_os_error());
+        }
+    }
+}
+
+const HECTONANOSECS_IN_SEC: i64 = 10_000_000;
+const HECTONANOSEC_TO_UNIX_EPOCH: i64 = 11_644_473_600 * HECTONANOSECS_IN_SEC;
+
+use crate::{Datelike, NaiveDateTime, Timelike};
+
+/// Chrono's internal wrapper of Window's [SYSTEMTIME][spec] structure that
+/// provides a variety of methods for interacting with Window's system calls
+/// and `SYSTEMTIME`.
+///
+/// [spec]: https://learn.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-systemtime
+pub(crate) struct WinSystemTime {
+    inner: SYSTEMTIME,
+}
+
+impl fmt::Debug for WinSystemTime {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("WinSystemTime")
+            .field("year", &self.inner.wYear)
+            .field("month", &self.inner.wMonth)
+            .field("dayOfWeek", &self.inner.wDayOfWeek)
+            .field("day", &self.inner.wDay)
+            .field("hour", &self.inner.wHour)
+            .field("minute", &self.inner.wMinute)
+            .field("second", &self.inner.wSecond)
+            .field("ms", &self.inner.wMilliseconds)
+            .finish()
+    }
+}
+
+impl WinSystemTime {
+    /// Creates a new empty `WinSystemTime`.
+    pub(crate) fn new() -> Self {
+        let st = SYSTEMTIME {
```

Comment:
I don't like that we're initializing `SYSTEMTIME` with invalid values here. I think it would be maybe more obvious to use `MaybeUninit` in the functions that need it.